### ISSUE TITLE
Support collecting resource statistics from SymbiFlow related tools

### ISSUE THREAD

**Issue by mithro** [olofk/edalize#214]
@olofk recently posted https://twitter.com/OlofKindgren/status/1354178191431245824?s=20
> I finally got around to merge a really cool Edalize feature. If you install edalize with the optional reporting module (e.g. `pip install --user -e ".[reporting]"`) you will get access to really powerful resource and timing statistics through Edalize. Check it out!
> ![image](https://user-images.githubusercontent.com/21212/106064559-ec4d5880-60ae-11eb-8679-aa79a1b1cf29.png)

It would be good to make sure this works with the SymbiFlow tooling including;
 - Yosys --> nextpnr
 - Yosys --> vpr

**Comment by mithro** [comment]
FYI - @acomodi / @kgugala

**Comment by olofk** [comment]
Yes! Definitely. Would love to have that. I'm hoping someone might be able to help out here. Happy to do it myself but I just have too much on my plate right now

**Comment by syed-ahmed** [comment]
I am willing to take this up! We have a use case for edalize and this feature!

To give a little more context, we are currently working on building a partial reconfiguration framework with the symbiflow backend. An example design on a defined rrgraph has a Makefile like [this](https://gist.github.com/syed-ahmed/c4e64bc4d5e58625418cbd25567403f6). I'm currently working on generating this Makefile with edalize. In addition, the reporting feature of edalize would enable us to do autotuning on several Yosys, VTR, etc. parameters.
